app/frontend/src/features/communities/discussions/Comment.tsx

```diff
@@ -0,0 +1,180 @@
+import { Card, CircularProgress, Typography } from "@material-ui/core";
+import { Skeleton } from "@material-ui/lab";
+import Avatar from "components/Avatar";
+import Button from "components/Button";
+import Markdown from "components/Markdown";
+import { useUser } from "features/userQueries/useUsers";
+import { Reply } from "pb/threads_pb";
+import { useEffect, useRef, useState } from "react";
+import { timestamp2Date } from "utils/date";
+import hasAtLeastOnePage from "utils/hasAtLeastOnePage";
+import isSafari from "utils/isSafari";
+import makeStyles from "utils/makeStyles";
+import { timeAgo } from "utils/timeAgo";
+
+import {
+  getByCreator,
+  LOAD_MORE_REPLIES,
+  REPLY,
+  UNKNOWN_USER,
+} from "../constants";
+import { useThread } from "../hooks";
+import CommentForm from "./CommentForm";
+
+const useStyles = makeStyles((theme) => ({
+  commentContainer: {
+    alignItems: "start",
+    columnGap: theme.spacing(2),
+    display: "grid",
+    gridTemplateAreas: `
+      "avatar content replyButton"
+    `,
+    gridTemplateColumns: "3rem 9fr 1fr",
+    gridTemplateRows: "auto",
+    padding: theme.spacing(2),
+    width: "100%",
+  },
+  commentContent: {
+    "& > * + *": {
+      marginBlockStart: theme.spacing(0.5),
+    },
+    display: "flex",
+    flexDirection: "column",
+    gridArea: "content",
+    marginInlineStart: theme.spacing(1),
+  },
+  replyButton: {
+    gridArea: "replyButton",
+    placeSelf: "end",
+  },
+  avatar: {
+    height: "3rem",
+    gridArea: "avatar",
+    width: "3rem",
+  },
+  nestedCommentsContainer: {
+    "& > * + *": {
+      marginBlockStart: theme.spacing(2),
+    },
+    display: "flex",
+    flexDirection: "column",
+    marginBlockStart: theme.spacing(2),
+    marginInlineStart: theme.spacing(3),
+    "&": {
+      marginInlineStart: `clamp(${theme.spacing(2)}, 5vw, ${theme.spacing(5)})`,
+    },
+  },
+  loadMoreRepliesButton: {
+    alignSelf: "center",
+  },
+}));
+
+export const COMMENT_TEST_ID = "comment";
+
+interface CommentProps {
+  comment: Reply.AsObject;
+  topLevel?: boolean;
+}
+
+export default function Comment({ topLevel = false, comment }: CommentProps) {
+  const classes = useStyles();
+  const { data: user, isLoading: isUserLoading } = useUser(
+    comment.authorUserId
+  );
+
+  const {
+    data: comments,
+    fetchNextPage,
+    hasNextPage,
+    isLoading: isCommentsLoading,
+    isFetching: isCommentsFetching,
+    isFetchingNextPage,
+  } = useThread(comment.threadId, { enabled: topLevel });
+  const isCommentsRefetching = !isCommentsLoading && isCommentsFetching;
+  const showLoadMoreButton = topLevel && hasNextPage;
+
+  const [showCommentForm, setShowCommentForm] = useState(false);
+  const commentFormRef = useRef<HTMLFormElement>(null);
+
+  useEffect(() => {
+    if (showCommentForm && commentFormRef.current) {
+      if (isSafari()) {
+        commentFormRef.current.scrollIntoViewIfNeeded(true);
```

Comment:
Safari doesn't support smooth scrolling AND scrolling element to centre of the screen even with the polyfill, but calling `scrollIntoViewIfNeeded(true)` does it but without smooth scrolling. I thought for Safari only, it's worth giving up smooth scrolling for this, otherwise the input always scroll to top (default option) of screen and clips off the first half of the input field.

Reply:
Also since I'm not measuring anything and probably because of the 1-2s delay caused by the animation revealing the input, I managed to get away with using `useEffect` to do this rather than with `useLayoutEffect` :D

Reply:
Maybe we need a better polyfill - does this one work? https://github.com/magic-akari/seamless-scroll-polyfill

Reply:
Haven't tried yet but from looking at the code, it looks more promising and comprehensive than the original polyfill we are using now (which this is a fork of!)

Reply:
That other polyfill worked like a charm, nice find!